src/Navigation.ts

```diff
@@ -0,0 +1,153 @@
+﻿class Navigation {
+	private table: Table;
+
+	private rowNum: number;			// The first Id to be fetched.
+	private numToFetch: number;		// The last Id to be fetched.
+	private searchedId: number;
+	private maxRecords: number;
+
+	private headerRowHeight = 23;
+	private bodyMargin = 8;
+	private searchBarHeight = 52;
+	private rowHeight = 24;
+
+	constructor(newTable: Table) {
+		this.table = newTable;
+		this.rowNum = 0;
+		this.numToFetch = 0;
+		this.searchedId = -1;
+		this.maxRecords = 0;
+	}
+
+	/**
+	 * Create a bold text node and an input field that will search through the
+	 * data.
+	 */
+	createSearchField() {
+		let bold = document.createElement('b');
+		let title = document.createTextNode("Search ID:");
+		bold.appendChild(title);
+		this.table.getMainTable().appendChild(bold);
+
+		let searchField = document.createElement("input");
+		searchField.id = "search";
+		searchField.type = 'number';
+		searchField.min = "0";
+		searchField.oninput = () => {
+			this.search();
+		}
+		this.table.getMainTable().appendChild(searchField);
+	}
+
+	/**
+	 * Create buttons for moving through the data one row at a time and
+	 * add them to the footer.
+	 */
+	createNavigationArrows() {
+		let footer = document.getElementById('mainFooter');
+
+		let downButton = document.createElement('button');
+		downButton.onclick = () => {
+			this.moveDown();
+		}
+		let imgDown = document.createElement('img');
+		imgDown.setAttribute('src', "icons/button_down.png");
+		downButton.appendChild(imgDown);
+		footer.appendChild(downButton);
+
+		let upButton = document.createElement('button');
+		upButton.onclick = () => {
+			this.moveUp();
+		}
```

Comment:
if callback is a single function or anon function calling 1 function, you can put on one line.